test/prob/discrete_range/discrete_range_ccdf_log_test.hpp

```diff
@@ -0,0 +1,65 @@
+// Arguments: Ints, Ints, Ints
+#include <stan/math/prim.hpp>
+
+using stan::math::var;
+using std::vector;
+
+class AgradCcdfLogDiscreteRange : public AgradCcdfLogTest {
+ public:
+  void valid_values(vector<vector<double>>& parameters,
+                    vector<double>& ccdf_log) {
+    vector<double> param(3);
+
+    param[0] = 3;  // y
+    param[1] = 1;  // lower
+    param[2] = 5;  // upper
+    parameters.push_back(param);
+    ccdf_log.push_back(-0.916290731874154995573);  // expected ccdf_log
+
+    param[0] = 9;   // y
+    param[1] = 5;   // lower
+    param[2] = 15;  // upper
+    parameters.push_back(param);
+    ccdf_log.push_back(-0.606135803570315601085);  // expected ccdf_log
+
+    param[0] = 0;   // y
+    param[1] = -4;  // lower
+    param[2] = 5;   // upper
+    parameters.push_back(param);
+    ccdf_log.push_back(-0.693147180559945286227);  // expected ccdf_log
+  }
+
+  void invalid_values(vector<size_t>& /*index*/, vector<double>& /*value*/) {
+    // y
```

Comment:
These are going to need the out-of-support tests.

In general, this shouldn't have three lines of comments double spaced.  A simple `// all values valid` comment is better because it doesn't look like someone forgot to fill in values here.

Reply:
I've added prim test for out-of-support values. The invalid_values section of distribution test doesn't let you express that easily, they are designed to capture values that are always invalid (say a negative rate parameter in a Poisson distribution). The formatting follows what's done in all other tests.